department-of-veterans-affairs/caseflow
# Rebuild Team Management Page

Connects [CASEFLOW-1968](https://vajira.max.gov/browse/CASEFLOW-1968)

### Description
This rebuilds the `TeamManagement` and related components to enable more functionality, better UX, and more.

### Acceptance Criteria
- [x] Team names that aren't editable display as wrapping text
- [x] Radio buttons for priority push toggle are replaced with dropdown with all/amaOnly/unavailable options
- [x] Add dropdown for requested push distribution with all/amaOnly options
- [x] Show status indicator for loading & saved states upon updating an org
- [x] [Recently added](https://github.com/department-of-veterans-affairs/caseflow/pull/16677) "VHA Regional Offices" section still shows up
- [x] JudgeTeam dropdowns are disabled unless user is DVC

### Testing Plan
1. Log in as `TEAM_ADMIN` user
2. Select "Caseflow Team Management" from user dropdown
3. Confirm page looks like design but that the dropdowns for distribution are disabled
4. Log in as DVC user
5. Select "Caseflow Team Management" from user dropdown
6. Confirm you see editable judge teams
7. Confirm you see saving indicator when making updates
8. Confirm DB fields actually get updated



### User Facing Changes

 BEFORE|AFTER
 ---|---
![Screen Shot 2021-09-03 at 11 00 59](https://user-images.githubusercontent.com/2581274/132026720-3a4f78a6-0c34-4409-9942-44172be23ae3.png) | ![Screen Shot 2021-09-03 at 11 00 44](https://user-images.githubusercontent.com/2581274/132026764-291e9716-550b-42c8-bd15-8999f1dd0cbe.png)
![Screen Shot 2021-09-03 at 11 04 23](https://user-images.githubusercontent.com/2581274/132027238-b96b0e0e-3142-4d00-889f-ef3b14edf611.png) | ![Screen Shot 2021-09-03 at 11 05 13](https://user-images.githubusercontent.com/2581274/132027273-4c538be8-213f-4b43-b63a-56a6060295ec.png)



### Storybook Story
*For Frontend (Presentationa) Components*
